[18.0][MIG] l10n_it_intrastat

tafaRU opened the pull request.

Commit:
porting l10n_it_intrastat to 11

Commit:
rinominati e spostati file per pulizia

Commit:
[MIG] l10n_it_intrastat: Migration to 12.0

Commit:
[FIX] Documentation and translation review

sbidoul committed:
Translated using Weblate (Italian)

Currently translated at 0.4% (42 of 10990 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_intrastat/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 14.5% (1595 of 10982 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_intrastat/it/

primes2h committed:
[FIX] translation source terms and readme

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 14.5% (1595 of 10982 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_intrastat/it/

sbidoul committed:
l10n_it_intrastat: Manual msgmerge

Investigating an issue reported by Sergio Zanchetta.

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 14.5% (1593 of 10990 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_intrastat/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 14.6% (1601 of 10988 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_intrastat/it/

Galoperd committed:
Added translation using Weblate (Spanish)

Galoperd committed:
Translated using Weblate (Spanish)

Currently translated at 0.5% (56 of 10988 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_intrastat/es/

Commit:
merge PR 1264 1717 1728 1737 1739

sergiocorato committed:
[12.0][FIX] where delete invoice intrastat lines has to be deleted too

SimoRubi committed:
[FIX] l10n_it_intrastat: Statement section not computed correctly, can't adapt type 'account.intrastat.transaction.nature' +  test

SimoneVagile committed:
Fix https://github.com/OCA/l10n-italy/issues/2058

eLBati committed:
IMP l10n_it_intrastat moving intrastat fields to dedicated tab

eLBati committed:
IMP l10n_it_intrastat category form: narrow fields

eLBati committed:
ADD name_get & name_search to report.intrastat.code to also show description

eLBati committed:
Do not compute_intrastat_lines if invoice is not subjected to intrastat

SimoRubi committed:
[IMP] l10n_it_intrastat: black, isort, prettier

SimoRubi committed:
[MIG] l10n_it_intrastat: Migration to 14.0
Pre-commit issues:
l10n_it_intrastat/models/account.py:304:5: C901 'AccountInvoice.compute_intrastat_lines' is too complex (17)
l10n_it_intrastat/models/account.py:540: [C8108(method-compute), AccountInvoiceIntrastat] Name of compute method should start with "_compute_"
l10n_it_intrastat/models/account.py:170:96: B950 line too long (95 > 80 characters)
l10n_it_intrastat/models/account.py:176:100: B950 line too long (99 > 80 characters)
l10n_it_intrastat/models/account.py:189:92: B950 line too long (91 > 80 characters)
l10n_it_intrastat/models/account.py:303:43: E741 ambiguous variable name 'l'
l10n_it_intrastat/models/account.py:373:21: B007 Loop control variable 'key' not used within the loop body. If this is intended, start the name with an underscore.
l10n_it_intrastat/models/account.py:380:25: B007 Loop control variable 'key' not used within the loop body. If this is intended, start the name with an underscore.
l10n_it_intrastat/models/account.py:397:17: B007 Loop control variable 'key' not used within the loop body. If this is intended, start the name with an underscore.
l10n_it_intrastat/models/account.py:409:13: E741 ambiguous variable name 'l'
l10n_it_intrastat/tests/test_intrastat.py:74:35: E741 ambiguous variable name 'l'

Migrate old invoice fields and fix tests

SimoRubi committed:
Add migration for v12:
- views do no more inherit from anything, otherwise following error might occur:
  Element '<tree string="Codes">' cannot be located in parent view

  View name: views.tree.report.intrastat.code
  Error context:
   view: ir.ui.view(2275,)
   xmlid: view_tree_report_intrastat_code
   view.model: report.intrastat.code
   view.parent: ir.ui.view(2269,)
   file: /path/to/odoo/addons/l10n_it_intrastat/views/intrastat.xml

- intrastat codes might have been moved by enterprise module account_intrastat

SimoRubi committed:
l10n_it_intrastat: Move size constraint of fields to web client.
Otherwise the following warning is raised:
unknown parameter 'size', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it

TheMule71 committed:
[REF] l10n_it_intrastat: Cleaner computation of statement section

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 14.5% (1601 of 10988 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_intrastat/it/

sergiocorato committed:
[IMP] intrastat 2022

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 14.5% (1603 of 10988 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_intrastat/it/

TheMule71 committed:
[FIX] l10n_it_intrastat: fix test tags

GSLabIt committed:
* l10n_it_instrastat, fix subject to intrastat when invoicing from sales order

GSLabIt committed:
[14.0][FIX] l10n_it_instrastat, account move: fix subject to intrastat on partner change. Better intrastat handling when invoice created from po,so.

michelerusti committed:
[14.0][IMP] l10n_it_instrastat - add option to omit weight from declaration

michelerusti committed:
[14.0][FIX]- l10n_it_intrastat_statement - weight_kg does not have to be always 1 if not > 0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10995 of 10995 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_intrastat/it/

TheMule71 committed:
[14.0][FIX] l10n_it_intrastat: migration leftover

mymage committed:
Translated using Weblate (Italian)

Currently translated at 99.9% (10994 of 10995 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_intrastat/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 99.9% (10994 of 10995 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10995 of 10995 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_intrastat/it/

Borruso committed:
[MIG] l10n_it_intrastat: Migration to 16.0

PicchiSeba committed:
[IMP][l10n_it_intrastat] add some missing intrastat codes

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10994 of 10994 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 99.9% (10994 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

Borruso committed:
[FIX] l10n_it_intrastat: fix nolabel view

SirAionTech committed:
Updated from template

SirAionTech committed:
[REF] l10n_it_intrastat: Update repo template

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11004 of 11004 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

eLBati committed:
FIX l10n_it_intrastat: multi company for account.invoice.intrastat

Otherwise, going to Contabilità > Righe intrastat, user would see data of every company

eLBati committed:
FIX l10n_it_intrastat avoid using today for EUR amount

Steps:

 - Install intrastat and configure products
 - Configure USD rate at 31/01 = 1,1
 - Configure USD rate at 01/02 = 1,5
 - In date > 31/01 create invoice with date = 31/01
 - Confirm and create intrastat statement

Observed:

Instrastat statement section contains EUR amount converted with 1,5 rate

Desired:

Instrastat statement section contains the real EUR amount

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11005 of 11005 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

eLBati committed:
[FIX] l10n_it_intrastat: compute weight in case of variants

Steps:

 - Activate variants
 - Configure a product with more than 1 variant
 - Set weights in variants
 - Create intrastat sale invoice
 - Add a product variant with a specific weight
 - Compute intrastat lines

Observed: weight is 0 in intrastat line

Borruso committed:
[ADD] l10n_it_intrastat: added intrastat country origin on product and category

Borruso committed:
[REF] l10n_it_intrastat: removed superfluous browse()

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11005 of 11005 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

Borruso committed:
[ADD] l10n_it_intrastat: added company rule on intrastat line

eLBati committed:
[FIX] l10n_it_intrastat using invoice ref
When reference (ref) is set, it must be used as invoice number

SirAionTech committed:
[FIX] l10n_it_intrastat: Allow overrides to return an action
For instance, module `account_financial_risk` needs to return an action when an invoice is posted

SirAionTech committed:
[COV] l10n_it_intrastat: compute weight from variants

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11006 of 11006 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_intrastat
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_intrastat/it/

tafaRU committed:
[IMP] l10n_it_intrastat: pre-commit auto fixes

tafaRU committed:
[MIG] l10n_it_intrastat: Migration to 18.0

tafaRU commented:
/ocabot migration l10n_it_intrastat

tafaRU commented:
/ocabot rebase

MaurizioPellegrinet reviewed (changes requested):
Manca sul prodotto il codice "nomenclatura combinata", non è visibile.
Sembra che la chiave (era good è diventata goods?) sia cambiata
![image](https://github.com/user-attachments/assets/df72795d-fdf7-4bd6-8b0d-69a57acabd48)

tafaRU commented:
@MaurizioPellegrinet puoi riprovare grazie.

MaurizioPellegrinet reviewed (approved):
Test funzionale: OK

eLBati commented on l10n_it_intrastat/models/account.py:
```diff
@@ -0,0 +1,681 @@
+class AccountFiscalPosition(models.Model):
+    _inherit = "account.fiscal.position"
+
+    intrastat = fields.Boolean(string="Subject to Intrastat")
```
Per compatibilità con https://github.com/OCA/intrastat-extrastat/blob/f66546a05b65c86882a94359015fd8fa66c11bb4/intrastat_base/models/account_fiscal_position.py#L12 e anche eventualmente altri, converrebbe rinominare i campi con prefisso `l10n_it`

tafaRU replied:
@eLBati 
> anche eventualmente altri

potresti essere più preciso? :)

tafaRU replied:
fatto in https://github.com/OCA/l10n-italy/pull/4718/commits/3b2444f2cd857be34d94e1a89fd18810a1de8ac5

tafaRU committed:
[IMP] l10n_it_intrastat: rename fields

This is due to avoid conflict with https://github.com/OCA/intrastat-extrastat/blob/f66546a05b65c86882a94359015fd8fa66c11bb4/intrastat_base/models/account_fiscal_position.py#L12

eLBati commented:
Abbiamo deciso di modificare solo i nomi dei campi che sappiamo avere problemi. Gli altri ce li possiamo tenere come sono.

Ovviamente per i nuovi moduli useremo il prefisso `l10n_it_oca`

eLBati reviewed (approved):
LGTM

eLBati commented:
/ocabot merge nobump
